plugins/spidmx/SpiDmxPlugin.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * SpiDmxPlugin.h
+ * This looks for possible SPI devices to instanciate and is managed by OLAD.
+ * Copyright (C) 2017 Florian Edelmann
+ */
+
+#ifndef PLUGINS_SPIDMX_SPIDMXPLUGIN_H_
+#define PLUGINS_SPIDMX_SPIDMXPLUGIN_H_
+
+#include <set>
+#include <string>
+#include <vector>
+
+#include "olad/Plugin.h"
+#include "ola/plugin_id.h"
+
+#include "plugins/spidmx/SpiDmxDevice.h"
+
+namespace ola {
+namespace plugin {
+namespace spidmx {
+
+class SpiDmxPlugin : public Plugin {
+ public:
+  explicit SpiDmxPlugin(PluginAdaptor *plugin_adaptor)
+      : Plugin(plugin_adaptor),
+        m_plugin_adaptor(plugin_adaptor) {
+  }
+
+  ola_plugin_id Id() const { return OLA_PLUGIN_EXPERIMENTAL; }
```

Comment:
Can you define a new OLA_PLUGIN_SPIDMX in https://github.com/OpenLightingProject/ola/blob/master/common/protocol/Ola.proto and set the value to 10000 for now, then we'll confirm an actual value when it gets merged. But doing it that way will increase the chances we'll remember to do so!

Reply:
When `make`ing, I get the following error:

```
"ola.proto.OLA_PLUGIN_EXPERIMENTAL" uses the same enum value as "ola.proto.OLA_PLUGIN_SPIDMX". If this is intended, set 'option allow_alias = true;' to the enum definition.
```

Should I add the suggested option somewhere (where?) or just set the constant to 9999?

Reply:
Interestingly this only happens on the OS X builds, which I think use a later protobuf.

As you can probably tell, we've only been through this particular workflow a few times with PRs. Essentially what we're trying to do initially is stopping you grabbing 23 for example and someone else having it too, as well as ensuring your code doesn't just use the OLA_PLUGIN_EXPERIMENTAL constant. I guess we could either comment out OLA_PLUGIN_EXPERIMENTAL (which will kill your same enum value error, or perhaps just change the comment there to say an ID greater than OLA_PLUGIN_EXPERIMENTAL, and allow a free for all up there, knowing they'll get updated before merging.

Reply:
Do we need the OLA_PLUGIN_EXPERIMENTAL then at all? The comment could just say that all IDs above 10000 are experimental and used during development until a fixed IDs is assigned.

Reply:
Yeah you're right, we probably don't. Hopefully that will make it less confusing for people too. Do you fancy trying to update the words for this, or would you rather I did it?

Reply:
I've made an attempt in b8ecf02. Is this okay?